### ISSUE TITLE
Regexp: violation pointing to line 0 while other Checks count first line as 1

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#14064]
https://checkstyle.org/checks/regexp/regexp.html#Regexp

problem is detected at https://github.com/checkstyle/checkstyle/pull/14057 and blocks it as test suite can not accept line 0 as expected. 

```
$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="Regexp">
      <property name="format" value="// This code is copyrighted"/>
    </module>
  </module>
</module>

$ cat Test.java 
// comment
public class TestClass {
}

$ java -jar checkstyle-10.12.4-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /var/tmp/Test.java:0: Required pattern '// This code is copyrighted' missing in file. [Regexp]
Audit done.
Checkstyle ends with 1 errors.

```

example for other Check that shows that line number is 1
```
$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
      <module name="FileLength">
        <property name="max" value="1"/>
      </module>
</module>

$ cat Test.java 
// comment
public class TestClass {
}

$ java -jar checkstyle-10.12.4-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /var/tmp/Test.java:1: File length is 3 lines (max allowed is 1). [FileLength]
Audit done.
Checkstyle ends with 1 errors.
```

prove in code that line length starts from 1: https://github.com/checkstyle/checkstyle/blob/55097468aa37d890b0e7b89ac2d27d08c16aee59/src/main/java/com/puppycrawl/tools/checkstyle/api/FileText.java#L279-L280

Expected:
`[ERROR] /var/tmp/Test.java:1: Required pattern '// This code is copyrighted' missing in file. [Regexp]`

**Comment by romani** [comment]
I think it should be "bug" and we consider breaking compatibility for config in most cases.
